cmd/config-reloader/main.go

```diff
@@ -73,6 +74,35 @@ func main() {
 		collectors.NewProcessCollector(collectors.ProcessCollectorOpts{}),
 	)
 
+	var cfgDirs []reloader.CfgDirOption
+	if *configDir != "" {
+		cfgDirs = append(cfgDirs, reloader.CfgDirOption{
+			Dir:       *configDir,
+			OutputDir: *configDirOutput,
+		})
+	}
+
+	newReloader := func(reg prometheus.Registerer, reloadURL *url.URL) *reloader.Reloader {
+		return reloader.New(
+			logger,
+			reg,
+			&reloader.Options{
+				ReloadURL:     reloadURL,
+				CfgFile:       *configFile,
+				CfgOutputFile: *configFileOutput,
+				CfgDirs:       cfgDirs,
+				WatchedDirs:   watchedDirs,
+				// There are some reliability issues with fsnotify picking up file changes.
```

Comment:
fsnotify works fine if you set it up to [add a new watch every time it detects a remove](https://github.com/kubernetes-sigs/controller-runtime/blob/main/pkg/certwatcher/certwatcher.go#L184) (which is how edits end up showing up in Kubernetes volumes).

That said, polling is probably fine, too.

Reply:
Suggestion: Remove this part of the comment.